From 99aa62729cbf1401c536fc7dfe69b90d55dde321 Mon Sep 17 00:00:00 2001
From: Chris Morgan <macromorgan@hotmail.com>
Date: Wed, 2 Feb 2022 10:42:00 -0600
Subject: [PATCH 1372/2201] overlays: Add rotate property to vc4-kms-dpi-panel

Allow a user to specify the panel rotation in devicetree as 0, 90,
180, or 270 by setting a parameter.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
---
 arch/arm/boot/dts/overlays/README           | 2 ++
 arch/arm/boot/dts/overlays/vc4-kms-dpi.dtsi | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 6a8d5c5f82c9..69cb31788b1a 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -3656,6 +3656,7 @@ Params: clock-frequency         Display clock frequency (Hz)
         backlight-def-brightness
                                 Set the default brightness. Normal range 1-16.
                                 (default 16).
+        rotate                  Display rotation {0,90,180,270} (default 0)
 
 
 Name:   vc4-kms-dpi-panel
@@ -3683,6 +3684,7 @@ Params: at056tn53v1             Enable an Innolux 5.6in VGA TFT
         backlight-def-brightness
                                 Set the default brightness. Normal range 1-16.
                                 (default 16).
+        rotate                  Display rotation {0,90,180,270} (default 0)
 
 
 Name:   vc4-kms-dsi-7inch
diff --git a/arch/arm/boot/dts/overlays/vc4-kms-dpi.dtsi b/arch/arm/boot/dts/overlays/vc4-kms-dpi.dtsi
index 02d7f7fb109c..f78fa48b19f9 100644
--- a/arch/arm/boot/dts/overlays/vc4-kms-dpi.dtsi
+++ b/arch/arm/boot/dts/overlays/vc4-kms-dpi.dtsi
@@ -10,6 +10,7 @@
 		target-path = "/";
 		__overlay__ {
 			panel: panel {
+				rotation = <0>;
 				port {
 					panel_in: endpoint {
 						remote-endpoint = <&dpi_out>;
@@ -105,5 +106,6 @@
 		backlight-pwm-gpio = <&pwm_pins>, "brcm,pins:0";
 		backlight-pwm-func = <&pwm_pins>, "brcm,function:0";
 		backlight-def-brightness = <&backlight_pwm>, "default-brightness-level:0";
+		rotate = <&panel>, "rotation:0";
 	};
 };
-- 
2.17.1

